pkg/manifests/manifests.go

```diff
@@ -831,6 +831,10 @@ func (f *Factory) PrometheusOperatorDeployment() (*appsv1.Deployment, error) {
 		return nil, err
 	}
 
+	if len(f.config.PrometheusOperatorConfig.NodeSelector) > 0 {
```

Comment:
Just curious, why do the `if` check at all?

Reply:
(I am open to suggest we have an overwrite-always mode, but then we should change it consistently.

Reply:
:+1: